beacon_node/src/cli.rs

```diff
@@ -316,22 +316,25 @@ pub fn cli_app<'a, 'b>() -> App<'a, 'b> {
         .arg(
             Arg::with_name("http-address")
                 .long("http-address")
+                .requires("http")
```

Comment:
Hey sorry @jxs, I've just realized that http api can be enabled via `--staking` or `--gui` as well, so `--http` is not strictly necessary, so this may be a breaking change for the beacon node, and we might want to leave this unchnaged?

https://github.com/sigp/lighthouse/blob/d61f5071849723f56c4cdd02e8a4f0c6100d1543/beacon_node/src/config.rs#L88-L91

https://github.com/sigp/lighthouse/blob/d61f5071849723f56c4cdd02e8a4f0c6100d1543/beacon_node/src/config.rs#L810-L813

Reply:
btw I discovered this from the failing `doppelganger` tests, it uses `--staking` instead of `--http`

Reply:
thanks for digging it Jimmy! I updated with a `ArgGroup` for `http`, `gui` and `staking`, when one of them is missing the following error is displayed:
```
error: The following required arguments were not provided:
    <--http|--gui|--staking>
```
Do you think it's clear enough? If not I can revert, meanwhile the `doppelganger` tests are still failing, can you access the logs? I am able to run the `beacon_node` for them locally but get a jwt validation error instead

Reply:
ok managed to fix the `doppelganger protection tests` and it seems to me it's mo longer a breaking change since all the flags still work as only one of the three is required, but ptal Jimmy